apache/trafficcontrol
# Prevents line breaks in delivery service remapText field (UI and API)

## What does this PR (Pull Request) do?

Rejects delivery service remapText with line breaks as it will result in a malformed remap.config file.

- [x] This PR fixes #4298 

## Which Traffic Control components are affected by this PR?

- Traffic Ops
- Traffic Portal

## What is the best way to verify this PR?
1. Run the [UI tests](https://github.com/apache/trafficcontrol/tree/master/traffic_portal/test/end_to_end)
2. Run the API tests
3. Manually verify in TP by trying to enter a line break in the `Raw Remap Text` delivery service form field for a DNS, HTTP or ANY_MAP delivery service.
4. Attempt to create or update a delivery service using the API where remapText contains \n or \r. It should fail.

## If this is a bug fix, what versions of Traffic Control are affected?
- master (6b11bd9)
- 3.1.0
- 4.0.0 (RC1)


## The following criteria are ALL met by this PR
